core/embed/rust/src/ui/mod.rs

```diff
@@ -12,10 +12,20 @@ pub mod screens;
 #[macro_use]
 pub mod util;
 
-#[cfg(feature = "micropython")]
 pub mod layout;
 
 #[cfg(feature = "model_tr")]
 pub mod model_tr;
 #[cfg(feature = "model_tt")]
 pub mod model_tt;
+
+#[cfg(all(
+    feature = "model_t1",
+    not(feature = "model_tr"),
+    not(feature = "model_tt")
+))]
+pub use model_t1 as model;
+#[cfg(all(feature = "model_tr", not(feature = "model_tt")))]
+pub use model_tr as model;
+#[cfg(feature = "model_tt")]
+pub use model_tt as model;
```

Comment:
this is actually a really bad idea because the individual models don't expose even vaguely similar API. sorry I didn't catch it soojner.

Reply:
i understand and agree that components are a bad place to do this. What i need is a model independent (or maybe more precisely ui-type independent) way to call some specific screen, or in other words, we need an universal API for UI in rust.

I guess that means introducing a new layer.. ultimately i think this should be layouts, with (optimally single) micropython interface refactored out. But thats quite a bit of refactoring too

Reply:
how about this?

make a trait that has all you need (contents of `simplified.rs` basically, right?)
implement it for every model
make the common functions generic over an impl of the trait
expose all versions of the top-level function to C, with the different impls under different names
use C #defines to pick which one gets actually called

it's a new layer, yes, and we can start small with it and extend it to cover the micropython interface later